js/src/components/tree-select-control/index.js

```diff
@@ -338,19 +426,20 @@ const TreeSelectControl = ( {
 			event.preventDefault();
 		}
 
-		if ( inputControlValue ) return;
-
 		if ( ARROW_UP === event.key ) {
+			if ( ! filteredOptions.length ) return;
 			setFocused( getPreviousOption( focused ) );
 			event.preventDefault();
 		}
 
 		if ( ARROW_DOWN === event.key ) {
+			if ( ! filteredOptions.length ) return;
 			setFocused( getNextOption( focused ) );
 			event.preventDefault();
 		}
 
 		if ( BACKSPACE === event.key ) {
+			if ( inputControlValue ) return;
 			onChange( value.slice( 0, -1 ) );
```

Comment:
❓ 
If the focus is on a checkbox, the behavior of pressing the Backspace key is different when searching and non-searching.

![Kapture 2022-03-23 at 15 49 45](https://user-images.githubusercontent.com/17420811/159659864-a3f39b52-c9a9-491b-8180-8f69337ff6fd.gif)

Reply:
> ❓ If the focus is on a checkbox, the behavior of pressing the Backspace key is different when searching and non-searching.
> 
> ![Kapture 2022-03-23 at 15 49 45](https://user-images.githubusercontent.com/17420811/159659864-a3f39b52-c9a9-491b-8180-8f69337ff6fd.gif)

Yes, this is because when is there a search text the behaviour should be to delete text instead of deleting the tags. However, I removed the keyDown for that in the Parent component and move it to the Control component.